datadogfirehosenozzle/datadog_firehose_nozzle.go

```diff
@@ -269,6 +269,8 @@ func (d *DatadogFirehoseNozzle) PostMetrics() {
 		metricsMap[k] = v
 	}
 	totalMessagesReceived := d.totalMessagesReceived
+	// Reset the map
+	d.metricsMap = make(metrics.MetricsMap)
```

Comment:
It took me some time to understand if we were introducing a regression or not here. In the end, this seems legit. This code needs some more inline comments. Great catch actually 👍 💯 🚀 🐕